nifi-nar-bundles/nifi-standard-services/nifi-ssl-context-bundle/nifi-ssl-context-service/src/main/java/org/apache/nifi/ssl/StandardSSLContextService.java

```diff
@@ -185,14 +191,24 @@ public void onPropertyModified(PropertyDescriptor descriptor, String oldValue, S
             }
         }
 
-        results.addAll(validateStore(validationContext.getProperties(), KeystoreValidationGroup.KEYSTORE));
-        results.addAll(validateStore(validationContext.getProperties(), KeystoreValidationGroup.TRUSTSTORE));
+        final Map<PropertyDescriptor, String> properties = evaluateProperties(validationContext.getProperties(), validationContext);
+        results.addAll(validateStore(properties, KeystoreValidationGroup.KEYSTORE));
+        results.addAll(validateStore(properties, KeystoreValidationGroup.TRUSTSTORE));
 
         isValidated = results.isEmpty();
 
         return results;
     }
 
+    private Map<PropertyDescriptor, String> evaluateProperties(final Map<PropertyDescriptor, String> properties, final PropertyContext context) {
+        for (final PropertyDescriptor pd : properties.keySet()) {
+            if (pd.isExpressionLanguageSupported() && context.getProperty(pd).isSet()) {
```

Comment:
It seems the intent of pd.isExpressionLanguageSupported() is to select only the keystore and truststore properties as they are the only properties in which EL is supported. If that is true, suggest you future-proof this if statement by being explicit and set conditions for only those two properties.

Reply:
My thinking was that this use of the loop was more future proof than explicitly checking individual properties (doing so would mean more updates in future if anyone wanted to make other properties EL enabled)

But per my reply to @exceptionfactory below, changing the `if` may allow a slightly simpler api of using the `PropertyContext`, so I'll consider both those comments together when I next look at the code

Reply:
I had a look at this, but decided not to make the change (unless you feel strongly about this) - it would mean adding code similar to the following:

```java
private final static List<PropertyDescriptor> EL_PROPERTIES = Collections.unmodifiableList(Arrays.asList(KEYSTORE, TRUSTSTORE));
```

Although actually would need to be an overridable method so that `StandardRestrictedSSLContextService` could potentially provide different values in the future.

Doing this would introduce a new Collection/method for developers to maintain in the future if they added more Properties with EL support (or changed existing properties to allow EL) - as they're already specifying that as part of the `PropertyDescriptor`, it seems unnecessary (and less maintainable) to have another set of properties to maintain in one or more classes (same argument for putting things directly in the `if`, e.g. `if (KEYSTORE.equals(pd) || TRUSTSTORE.equals(pd)) { ... }` - I'd say that was even less maintainable as now the developer needs to know to look further through the code to understand what needs updating.